src/components/Comment.jsx

```diff
@@ -1,15 +1,60 @@
 import formatDate from "../utils/formatDate";
+import { useContext, useState, useEffect } from "react";
+import { UserContext } from "./contexts/User";
+import { deleteComment } from "../utils/api";
 
 const Comment = ({ comment }) => {
+  const { user, setUser } = useContext(UserContext);
+  const [canDelete, setCanDelete] = useState(false);
```

Comment:
You could refactor this, if the user exists that is also a truthy or falsy value for conditional rendering. Its not great to have a huge amount of state. Sometimes we can get the same result by looking at the data we already have, and if we _really can't_ do it without more state then thats ok at add it.